csfieldguide/templates/interactives/matrix-simplifier.html

```diff
@@ -0,0 +1,190 @@
+
+{% load i18n %}
+{% load static %}
+
+{% block html %}
+  <div class="container">
+    <h3 class='text-center'>{% trans "Matrix Simplifier" %}</h3>
+    <p class="text-center">{% trans "Try dragging the matrices and vectors below to change their order. Does this affect the output?" %}</p>
+    <div class="row">
+      <div class="col-md-6 col-sm-12 text-center mb-3">
+        <h5>{% trans "Matrices" %}</h5>
+        <div id="matrix-input-container" class="containers">
+          <div class="draggable content border rounded m-1 center-block">
+            <button id="close-matrix-0" type="button" class="close dismiss-eqtn" aria-label="Close">
+              <span aria-hidden="true">&times;</span>
+            </button>
+            <div id="matrix-0" class="matrix invisible" data-matrix-order=0>
+              \[
+                \begin{bmatrix}
+                cos(45)   &  0  &  sin(45) \\
+                0         &  1  &  0 \\
+                -sin(45)  &  0  &  cos(45) \\
+                \end{bmatrix}
+              \]
+            </div>
+          </div>
+      
+          <div class="draggable content border rounded m-1 center-block">
+            <button id="close-matrix-1" type="button" class="close dismiss-eqtn" aria-label="Close">
+              <span aria-hidden="true">&times;</span>
+            </button>
+            <div id="matrix-1" class="matrix invisible" data-matrix-order=1>
+              \[
+                \begin{bmatrix}
+                10  &  0  &  0 \\
+                0   &  10 &  0 \\
+                0   &  0  &  10 \\
+                \end{bmatrix}
+              \]
+            </div>
+          </div>
+        </div>
+        <button id="add-matrix-btn" type="button" class="btn btn-primary d-none" data-toggle="modal" data-target="#matrix-modal">{% trans "Add matrix" %}</button>
+      </div>
+
+      <div class="col-md-6 col-sm-12 text-center mb-3">
+        <h5>{% trans "Vectors" %}</h5>
+        <div id="vector-input-container" class="containers">
+          <div class="draggable content border rounded m-1 center-block">
+            <button id="close-vector-0" type="button" class="close dismiss-eqtn" aria-label="Close">
+              <span aria-hidden="true">&times;</span>
+            </button>
+            <div id="vector-0" class="vector invisible" data-vector-order=0>
+              \[
+                \begin{bmatrix}
+                10 \\
+                0 \\
+                0 \\
+                \end{bmatrix}
+              \]
+            </div>
+          </div>
+        </div>
+        <button id="add-vector-btn" type="button" class="btn btn-primary d-none" data-toggle="modal" data-target="#vector-modal">{% trans "Add vector" %}</button>
+      </div>
+    </div>
+
+    <div class="row">
+      <h5 class="col-12 text-center">{% trans "Result" %}</h5>
+    </div>
+
+    <div class="row">
+      <div id="output-container" class="col-12 text-center invisible">
```

Comment:
I would be happier to implement this as part of the transform interactive rebuild so it can be tested correctly :) would you be ok with that? Good idea.

Reply:
That's fine, for now I think just make sure there is at least a way to access the result so you don't have to rebuild everything later